--- a/esp_lcd_st7701_mipi.c
+++ b/esp_lcd_st7701_mipi.c
@@ -65,7 +65,7 @@ esp_err_t esp_lcd_new_panel_st7701_mipi(const esp_lcd_panel_io_handle_t io, con
         ESP_GOTO_ON_ERROR(gpio_config(&io_conf), err, TAG, "configure GPIO for RST line failed");
     }

-    switch (panel_dev_config->color_space) {
+    switch (panel_dev_config->rgb_ele_order) {
     case LCD_RGB_ELEMENT_ORDER_RGB:
         st7701->madctl_val = 0;
         break;
